include/ppx/scene/scene_material.h

```diff
@@ -167,6 +170,12 @@ class Material
     virtual std::string GetIdentString() const = 0;
 
     virtual scene::VertexAttributeFlags GetRequiredVertexAttributes() const = 0;
+
+    // Returns true if material has paramters for shader
+    virtual bool HasParams() const { return false; }
```

Comment:
Should this be pure virtual?  May be easier to spot problems when a derived class doesn't implement this.

Reply:
I don't think that's necessary. My expectation is that if you're implementing a material, I will trust that you will know if it's going to use params, textures or both.

Reply:
If people feel really strongly about ADTs I'll change it.

Reply:
I'm in favor of the pure virtual: when building an interface, I'm in favor or making each non-universal function pure virtual. As the default behavior here is not always subset of the correct child behavior.

Reply:
I will change it to a pure virtual interface to keep you folks happy.

Reply:
Functions are now pure virtual.